staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -220,7 +220,7 @@ func fromUnstructured(sv, dv reflect.Value) error {
 	case reflect.Interface:
 		return interfaceFromUnstructured(sv, dv)
 	default:
-		return fmt.Errorf("unrecognized type: %v", dt.Kind())
+		return fmt.Errorf("can't convert value %#v of type %v -> %v", sv, st.Kind(), dt.Kind())
```

Comment:
don't serialize the actual value into error strings, that could be excessively large... the kind is the issue here

Reply:
I could easily see how `Expected int, got string` is super unhelpful though? Can't we provide the path to the filed?

Reply:
@liggitt 
I have dropped the value.

Not sure of how to get the path though.